### PR TITLE
feat: support specifying provider branch in verifications

### PR DESCRIPTION
**Pull request by mefellows**
Fixes https://github.com/pact-foundation/pact-js/issues/750 (for the `pact-core` package)

### CODE REVIEW

**Comment by mefellows** [comment]
Sorry, hit review too early Tim. I need to upgrade to the latest FFI and test it out as per the discussion in the #pact-rust channel recently.

**Comment by TimothyJones** [comment]
It's also worth diff'ing the C header file. I was thinking maybe we should check it in to make changes more obvious

**Comment by TimothyJones** [comment]
What is the status of this one?

**Comment by mefellows** [comment]
Ah, the status is that bumping the FFI version alone won't work, despite the flag being correct.

The issue is that the FFI mappings need to change a little with the newer n

See https://pact-foundation.slack.com/archives/CA2S7E6KC/p1637046326013500 and https://pact-foundation.slack.com/archives/CA2S7E6KC/p1637047542017900 for some context.

TL;DR - another version bump is required and then the two structs can just be referenced as opaque pointers (the implementation details are now hidden from us).

**Comment by mefellows** [comment]
`providerBranch` replaces this in current master. Closing defunct PR.
